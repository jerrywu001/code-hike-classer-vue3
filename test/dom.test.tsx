import '@testing-library/jest-dom';

describe('dom test', () => {
  test('dom', () => {
    const div = document.createElement('div');
    div.id = 'adm-mask';
    expect(div).not.toBeNull();
    expect(div).toBeDefined();
    expect(div).toBeInstanceOf(HTMLDivElement);

    document.body.appendChild(div);

    const mask = document.body.querySelector('#adm-mask') as HTMLDivElement;
    expect(mask).toMatchSnapshot();
    div.remove();
    expect(mask).not.toBeInTheDocument();
  });
});
