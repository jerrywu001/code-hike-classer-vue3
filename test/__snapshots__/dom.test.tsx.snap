// Vitest Snapshot v1

exports[`dom test > dom 1`] = `
<div
  id="adm-mask"
/>
`;
